projects/anti_scaling/distillation.py

```diff
@@ -158,34 +158,34 @@ def __init__(self, opt, shared=None):
 
         # Create teacher model
         if shared is None:
-            to_copy = {'no_cuda', 'model_parallel', 'fp16', 'fp16_impl'}
+            to_copy = {'no_cuda', 'model_parallel', 'fp16', 'fp16_impl', 'gpu'}
             override = {k: opt[k] for k in to_copy}
             override['datatype'] = 'train:evalmode'  # Don't initialize the optimizer
             teacher_agent = create_agent_from_model_file(opt['teacher_model'], override)
             self.teacher_model = teacher_agent.model
             assert teacher_agent.opt['n_heads'] == opt['n_heads']
-            if hasattr(self.teacher_model, 'module'):
-                self.teacher_model = self.teacher_model.module
```

Comment:
This new change seems surprising to me?

Reply:
Ah yes - my thinking for this change was that, if you set `self.teacher_model` to the `'module'` attribute of the teacher in the __init__ method of the distillation agent, the effect of `DistributedDataParallel` may not get applied during training, but I don't know offhand if that's actually the case. This shift in the placement of this code also echos the fact that `self.model.module` in TGA is referenced in `TGA._generate()`, not `TGA.__init__()`